translate.go

```diff
@@ -0,0 +1,119 @@
+package i18n
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/jibber_jabber"
+)
+
+var (
+	localeRegexp  string          = "^[a-z]{2}([_-][A-Z]{2}){0,1}$"
+	log                           = golog.LoggerFor("i18n")
+	fs            http.FileSystem = http.Dir("locale")
+	defaultLocale string          = "en_US"
+	defaultLang   string          = "en"
+	trMutex       sync.RWMutex
+	trMap         map[string]string
+)
+
+// T translates the given key into a message based on the current locale,
+// formatting the string using the supplied (optional) args. This method will
+// fall back to other locales if the key isn't defined for the current locale.
+// The search order (with examples) is as follows:
+//
+//   1. current locale        (zh_CN)
+//   2. lang only             (zh)
+//   3. default locale        (en_US)
+//   4. lang only of default  (en)
+//
+func T(key string, args ...interface{}) string {
+	trMutex.RLock()
+	defer trMutex.RUnlock()
+	s := trMap[key]
+	// Format string
+	if s != "" && len(args) > 0 {
+		s = fmt.Sprintf(s, args...)
```

Comment:
What I'm looking for in the test is an existing message with some format strings in there, like "A Test Message %s" that then gets a parameter passed to it.  [this](https://github.com/getlantern/i18n/pull/1/files#diff-0c2d284c965b4de2439149620ff07077R30) appears to be testing a missing message.